src/remote/remote.h

```diff
@@ -348,16 +348,11 @@ struct Rtr : public Firebird::GlobalStorage, public TypedHandle<rem_type_rtr>
 
 struct RBlobInfo
 {
-	bool	valid;
-	UCHAR	blob_type;
-	ULONG	num_segments;
-	ULONG	max_segment;
-	FB_UINT64	total_length;
-
-	RBlobInfo()
-	{
-		memset(this, 0, sizeof(*this));
-	}
+	bool	valid{};
+	UCHAR	blob_type{};
+	ULONG	num_segments{};
+	ULONG	max_segment{};
+	FB_UINT64	total_length{};
```

Comment:
These are simple types I would prefer `num_segment = 0` syntax because of better readability and explicit blob type constant for the same sake.

Reply:
C++ Core Guidelines: [ES.23: Prefer the {}-initializer syntax](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Res-list). I could replace it with `{false}`, `{0}` but as that is already the default value when using `{}`, that seems unnecessary to me.

Reply:
Not every guideline worth following. In this case narrowing surely cannot happen. And it is better not to make readers guess which blob type is default.

Reply:
Point taken for blob_type, but for the rest, I think this is explicit enough if you're familiar with the initializer syntax.

Reply:
I'm also against these `{}`. And the team never discussed about following C++ Core Guidelines.

Reply:
OK, what do you want then? Explicit list-initialization (`{ false }`/`{ 0 }`), or copy-initialization (`= false`/`= 0`). I did it primarily to get rid of that memset ([C.90: Rely on constructors and assignment operators, not memset and memcpy](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rc-memset)).

And re C++ Core Guidelines, given list-initializers (or initialization in general) are not mentioned in https://geosoft.no/development/cppstyle.html (which is referenced from https://www.firebirdsql.org/en/coding-style), I referenced a more complete style guide, maintained by people who designed C++ to get an impression of what desirable style is. I'm not saying it is the bible, or that we must follow it, I'm using it to explain _why_ I did something

Reply:
`name = <value>` is fine for me.

Reply:
> `name = <value>` is fine for me.

+1